apps/bestofjs-nextjs/src/components/search-palette/search-palette.tsx

```diff
@@ -54,11 +52,11 @@ export function SearchPalette() {
     onSelectSearchForText,
     onViewAllTags,
     open,
+    removeTag,
     searchQuery,
     selectedItem,
     setOpen,
   } = useSearchPaletteState();
-  const { currentTags, currentTagCodes, removeTag } = useSearchPaletteTags();
```

Comment:
So if I'm reading this correctly.

`useSearchPaletteTags` is removed and its functionality is included in `useSearchPaletteState`?

As I don't see any changes to both hooks in this PR the situation is unclear.

Reply:
The problem: before was we were calling 2 different hooks `useSearchPaletteState()` and `useSearchPaletteTags()` that in turn was calling `useSearchPaletteState()`, so the result was the two calls of `useSearchPaletteState()` when rendering, which created two different states and caused the bug (because there was not context or any mechanism to share the state involved, it's a common error when composing hooks!)

This is why I chose to remove  the hook `useSearchPaletteTags` and only use what is returned by `useSearchPaletteState` instead.

Reply:
Makes sense!